### PR TITLE
pass through cache_options

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/intake/filesystem_spec/issues/205

@hayesgb can you test this out with fsspec>=0.6? We don't have unit tests for gen1 AFAICT.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
cc @JoranDox.

**Comment by hayesgb** [comment]
Currently traveling for work, but I should be able to test it tonight.

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 azure-datalake-store>=0.0.46,<0.1
 azure-storage-blob>=2.1.0
```
Comment (hayesgb) [inline requirements.txt]: Need to pin this to azure-storage-blob<3.0  Microsoft is currently doing a large rewrite, and the version on PyPi jumped from 2.1.0 --> 12.0.

Comment (martindurant) [reply requirements.txt]: > the version on PyPi jumped from 2.1.0 --> 12.0.

! 
It must be six times better!

**Comment by TomAugspurger** [comment]
In general, do we want to set max pins?

**Comment by hayesgb** [comment]
That’s hilarious @martindurant.  

The only reason I would suggest it is to minimize the risk of breaking changes with major version changes in dependencies.

**Comment by hayesgb** [comment]
This seems to work just fine with fsspec=0.6.0 in Azure Datalake Gen1.

**Comment by martindurant** [comment]
Time to add to fsspec.registry.known_implementations?

**Comment by TomAugspurger** [comment]
Probably. Is that done on the fsspec or adlfs side?

Have you considered using entrypoints for this so that it can be done
without needing to import the package?

**Comment by martindurant** [comment]
in fsspec: https://github.com/intake/filesystem_spec/blob/master/fsspec/registry.py#L12 . Each package is only imported when required.

Yes, entry points would be a reasonable way to do this for the future.

**Comment by hayesgb** [comment]
What was the final decision on abfs as a protocol?  Everything is written for abfs, but an alternative was suggested.

**Comment by martindurant** [comment]
I vote "abs"

**Comment by hayesgb** [comment]
I’m good with either — my only hesitation is whether we should maintain consistency with the Azure HDFS specification, which is abfs[s]://

Thoughts?

**Comment by martindurant** [comment]
Is that format published somewhere? If yes, then I suppose it makes our decision for us.

**Comment by hayesgb** [comment]
It is.  I can send a link to the docs later, but it’s on the Apache site.

**Pull request merged**